File: photomap-core-api/photomap-common/dto/src/main/java/ua/in/photomap/common/photo/model/dto/PhotoGeoJsonDataDTO.java
package ua.in.photomap.common.photo.model.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class PhotoGeoJsonDataDTO {
    private CoordinatesDTO coordinates;
    private Long photoId;
    private String iconThumbUrl;
    private List<String> tags;
    private YearRangeDTO yearRange;
}

File: photomap-core-api/photo-api-python/dto/photo_geojson_data_dto.py
from pydantic import BaseModel, Field
from typing import List, Optional
from dto.photo_dto import CoordinatesDTO, YearRangeDTO

class PhotoGeoJsonDataDTO(BaseModel):
    coordinates: CoordinatesDTO = Field(description="Photo coordinates")
    photoId: int = Field(description="Photo ID")
    iconThumbUrl: Optional[str] = Field(default=None, description="Icon thumbnail URL")
    tags: List[str] = Field(description="Photo tags")
    yearRange: YearRangeDTO = Field(description="Year range") 